Automattic/wp-calypso
# Auto-fix ESLint rule react/jsx-curly-brace-presence and start enforcing it

**Note**: this PR should be merged into trunk after https://github.com/Automattic/wp-calypso/pull/69021 which clears some errors that get reported by CI in this PR due to it changing a lot of files. Once that PR is merged this one can rebase and the CI code check task should pass. 

Related to https://github.com/Automattic/wp-calypso/pull/68939

#### Proposed Changes

* Sometimes `react/jsx-curly-brace-presence` is flagged in code reviews and sometimes it isn't. For consistency, this PR proposes to enforce this rule and also benefit from the auto-fix feature of `eslint`. 

I opted for this rule config based on previous code review feedback:

![image](https://user-images.githubusercontent.com/6851384/195240504-bca2d897-2d79-4b1f-aeb2-4a06faeb93a1.png)

The rule homepage and full config options are [here](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-curly-brace-presence.md) 

#### Testing Instructions



* Review the files that were auto-fixed
* Checkout this branch and try to modify some JSX props to be like the **incorrect** format above. Verify they are caught by ESLint and can be auto-fixed. 

#### Pre-merge Checklist






Related to #
